src/components/DonationItemsTable/DonationsModal.jsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+import { useBackend } from '../../contexts/BackendContext';
+import PropTypes from 'prop-types';
+import { Input, Text, Modal, ModalContent, ModalHeader, ModalBody, ModalOverlay, ModalFooter, IconButton, Button} from '@chakra-ui/react';
+import {CloseIcon} from '@chakra-ui/icons';
+
+const DonationsModal = ({ isOpen, onClose, data, setData, currentPageNum }) => {
+  const { backend } = useBackend();
+  const [categoryData, setCategoryData] = useState('');
+  const [typeData, setTypeData] = useState('');
+  const [weightData, setWeightData] = useState('');
+  const [priceData, setPriceData] = useState('');
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        if (isOpen && data) {
+          setCategoryData(data['category']);
+          setTypeData(data['item_name']);
+          setWeightData(data['quantity_type']);
+          setPriceData(data['price']);
+        }
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+    getData();
+  }, [isOpen]);
+
+  const submitForm = async event => {
+    event.preventDefault();
+    if (!data){
+      await backend.post(`/value`, {
+        'itemName': typeData,
+        'quantityType': weightData,
+        'price': priceData,
+        'category': categoryData,
+      });
+    } else{
+      await backend.put(`/value/${data["item_id"]}`, {
+        'itemName': typeData,
+        'quantityType': weightData,
+        'price': priceData,
+        'category': categoryData,
+      });
+    }
+    const dataResponse = backend.get(
+      `/value/?itemsLimit=10&pageNum=${currentPageNum}`,
```

Comment:
make this do the get all GET request also. otherwise we have a problem where when you add the 11th item (or like if you have a full page and add another item), it won't be loaded into the tab until the page is reloaded.